### DISCUSSION TITLE
A Package Manifest should say what it includes

### DISCUSSION THREAD

**Discussion by shoulders** [joomla/joomla-cms#32515] [2. Cold Features]
I would like to see the package manifest have a section where it tell you what type of extensions are contained. This would be useful for JED automation.

Below are some examples, the tag name `include` could be renamed if needed but i used this JED entry as a basis for the name: 
https://extensions.joomla.org/extension/jomdirectory-cart/

```xml
<includes>
    <include>com</include>
    <include>plg</include>
    <include>mod</include>
    <include>tpl</include>
    <include>lib</include>
    <include>file</include>            
    <include>extspec</include>
    <include>bin</include>
</includes>
```
or
```xml
<includes>
    <include>component</include>
    <include>plugin</include>
    <include>module</include>
    <include>template</include>
    <include>library</include>
    <include>file</include>            
    <include>extension-specific</include>
    <include>binary</include>
</includes>
```

**Comment by shoulders** [comment]
thinking about it i might be able to parse

```xml
<filename type="plugin"    group="system"     id="kunena"      >plg_system_kunena_v5.2.2.zip</filename>
<filename type="plugin"    group="quickicon"  id="kunena"      >plg_quickicon_kunena_v5.2.2.zip</filename>
<filename type="library"    id="kunena">lib_kunena_v5.2.2.zip</filename>
<filename type="file"         id="kunena_media">kunena_media_v5.2.2.zip</filename>
<filename type="component" id="com_kunena"  >com_kunena_v5.2.2.zip</filename>
```

**Reply by brianteeman** [reply]
How would it be different and what would be the benefit

**Reply by shoulders** [reply]
I was looking at automating uploading and parsing extensions for my own projects system and I saw how `<tags>` were listed and thought that would be a good idea. After I typed out thie idea above I then figured out that I could parse the actual `<filename>` tags and build my own that way but could not close the idea.
